ts/IterableInbox.tsx

```diff
@@ -1,25 +1,58 @@
-'use strict'
-
-import React, { Component, useState } from 'react'
-import { Text, View, StyleSheet } from 'react-native'
-import IterableInboxMessageList from './IterableInboxMessageList'
-import IterableInboxEmptyState from './IterableInboxEmptyState'
-import sampleMessages from './sampleMessageData.js'
+'use strict';
+import React, { Component, useState } from 'react';
+import { Text, View, StyleSheet } from 'react-native';
+import IterableInboxMessageList from './IterableInboxMessageList';
+import IterableInboxEmptyState from './IterableInboxEmptyState';
+import IterableInboxMessageDisplay from './IterableInboxMessageDisplay';
+import sampleMessages from './sampleMessageData.js';
 
 const IterableInbox = () => {
-   const message = "Inbox";
+   const inboxTitle = "Inbox";
+   const [isDisplayMessage, setIsDisplayMessage] = useState(false);
+   const [selectedMessageIdx, setSelectedMessageIdx] = useState(0);
    const [messages, setMessages] = useState(sampleMessages);
 
+   let selectedMessage = messages[selectedMessageIdx];
+
+   function handleMessageSelect(index: number, messages: Array<any>) {
+      if(!messages[index].read) {
+         messages[index].read = true
+      }
+      setMessages([...messages]);
+      setIsDisplayMessage(true);
+      setSelectedMessageIdx(index);
+   }
+
+   function returnToInbox() {
+      setIsDisplayMessage(false);
+   }
+   
+   function displayMessage(message: {[key: string]: any}) {
+      return (
+         <IterableInboxMessageDisplay 
+            message={message}
+            returnToInbox={returnToInbox}
+         ></IterableInboxMessageDisplay>)
+   }
+
    function showMessageList() {
```

Comment:
Same comment as `displayMessage' here.

Reply:
I agree. I renamed the function to showMessageDisplay to align with showMessageList.